Rise-Vision/rise-vision-apps
# Save and Publish branding settings; branding colors directive

## Description
**Branding settings**
Clear out draft values on publish
Detect draft state based on missing draft values

Update preview to post branding info to template

Factor out blueprint and financial-license factories

**Branding colors directive**
Directive loads and saves factory data
Move more functionality to factories

## Motivation and Context
UI implementation for the Branding Epic

## How Has This Been Tested?
Tested functionality of both existing Templates and the Branding Example Template in both local and staging environments.
Branding settings are not shown for every Template, but only the `branding: true` templates, hence this change should not be visible to Users.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
**Manual Test**  As per the above
**Automated Test** Unit tests were updated for the change
**Monitoring** Will test in Production once released and monitor that no errors are occurring
**Rollback** Rollback can be done by reverting the PR and following the deployment process
**Documentation** Branding architecture doc: https://docs.google.com/document/d/1XZUcQ6D9oWjMxrllbZKZLCDV3D8W89OyqrUMwVjEp_8/edit#heading=h.qyglks5gphd1
**Internal Communication** Will notify interested parties once deployed

- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
None
